external/dockerfile_functions.sh

```diff
@@ -148,8 +162,6 @@ print_ubuntu_pkg() {
 # Select the ubuntu OS packages
 print_ubi_pkg() {
     local file=$1
-    local packages=$2
-
     echo -e "RUN dnf install -y ${packages} \\" \
```

Comment:
The variable `packages` was removed, but still used here. This might be the reason why `RUN dnf install -y    	&& dnf clean all` cannot catch any packages.
If some core functions in this PR are needed in urgent, as Sophia suggested, we can extract this PR to only address the core functions, then work on non-core functions, e.g. `generice_packages` in other PRs, combining them all in this PR makes it easy to break current build. Thanks.